### ISSUE TITLE
Getting no valid credential sources found for AWS Provider

### ISSUE THREAD

**Issue by gugaiz** [scalefactory/tfctl#14]
I am doing my initials test using the `example-profile` provided on the repo. I just changed the accounts in `tfctl.yml` to match the ones on my organization. I have setup the credentials using environment variables. When I run `tfctl --all -- init` I get

```
info: tfctl running
info: Using config: default
info: Working out AWS account topology
info: Merging configuration
info: account-production: Generating Terraform run directory
info: account-dev: Generating Terraform run directory
info: account-staging: Generating Terraform run directory
info: account-dev: Executing Terraform init
info: account-production: Executing Terraform init
info: account-staging: Executing Terraform init
info: account-staging: Initializing modules...
info: account-staging: 
info: account-staging: Initializing the backend...
error: account-staging: 
error: account-staging: Error: No valid credential sources found for AWS Provider.
error: account-staging:   Please see https://terraform.io/docs/providers/aws/index.html for more information on
error: account-staging:   providing credentials for the AWS Provider
error: account-staging: 
error: account-staging: 
info: account-production: Initializing modules...
info: account-production: 
info: account-production: Initializing the backend...
error: account-production: 
error: account-production: Error: No valid credential sources found for AWS Provider.
error: account-production:  Please see https://terraform.io/docs/providers/aws/index.html for more information on
error: account-production:  providing credentials for the AWS Provider
error: account-production: 
error: account-production: 
info: account-dev: Initializing modules...
info: account-dev: 
info: account-dev: Initializing the backend...
error: account-dev: 
error: account-dev: Error: No valid credential sources found for AWS Provider.
error: account-dev:   Please see https://terraform.io/docs/providers/aws/index.html for more information on
error: account-dev:   providing credentials for the AWS Provider
error: account-dev: 
error: account-dev: 
error: terraform failed with exit code: 1
```

**Comment by gugaiz** [comment]
FYI: If I go to `.tfctl/default/account-dev` and run `terraform init` it works

**Comment by gugaiz** [comment]
I think I found the issue. The problem is that you are getting rid of all env variables hen calling ENV.keep_if [here](https://github.com/scalefactory/tfctl/blob/master/lib/tfctl/executor.rb#L52), then terraform command does not have the correct env setup. PR #13 solves the problem for me

**Comment by word** [comment]
Thanks for the bug report.  This should be fixed in the latest release: 1.2.2.  Details in PR #13.
